psi4/src/psi4/dfmp2/mp2.cc

```diff
@@ -2820,6 +2822,222 @@ void RDFMP2::form_gradient()
         }  // end external
     }
 
+
+    // => Perturbation Gradient <= //
+    if(options_.get_bool("PERTURB_H")) {
+        timer_on("Grad: Perturbation");
+
+        double xlambda = 0.0;
+        double ylambda = 0.0;
+        double zlambda = 0.0;
+
+        std::string perturb_with = options_.get_str("PERTURB_WITH");
+        if (perturb_with == "DIPOLE_X")
+            xlambda = options_.get_double("PERTURB_MAGNITUDE");
+        else if (perturb_with == "DIPOLE_Y")
+            ylambda = options_.get_double("PERTURB_MAGNITUDE");
+        else if (perturb_with == "DIPOLE_Z")
+            zlambda = options_.get_double("PERTURB_MAGNITUDE");
+        else if (perturb_with == "DIPOLE") {
+            if(options_["PERTURB_DIPOLE"].size() !=3)
+                throw PSIEXCEPTION("The PERTURB dipole should have exactly three floating point numbers.");
+            xlambda = options_["PERTURB_DIPOLE"][0].to_double();
+            ylambda = options_["PERTURB_DIPOLE"][1].to_double();
+            zlambda = options_["PERTURB_DIPOLE"][2].to_double();
+        } else {
```

Comment:
I only kept the former as an option so that the update doesn't break existing inputs.  The manual and the code have deprecation warnings about `dipole_y` and `perturb_magnitude` so they should be removed eventually.